File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.java
class Solution {
    public int totalNumbers(int[] digits) {
        int[] count = new int[10];
        HashSet<Integer> set = new HashSet<>();
        int ans = 0;

        for (int digit : digits) {
            count[digit]++;
        }

        for (int i = 1; i <= 9; i++) {
            if (count[i] == 0)
                continue;
            count[i]--;

            for (int j = 0; j <= 9; j++) {
                if (count[j] == 0)
                    continue;
                count[j]--;

                for (int k = 0; k <= 9; k++) {
                    if (count[k] == 0)
                        continue;
                    if (k % 2 == 0) {
                        count[k]--;
                        int num = 100 * i + 10 * j + k;

                        if (!set.contains(num)) {
                            set.add(num);
                            ans++;
                        }
                        count[k]++;
                    }
                }
                count[j]++;
            }
            count[i]++;
        }
        return ans;
    }
}

File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.js
/**
 * @param {number[]} digits
 * @return {number}
 */
var totalNumbers = function (digits) {
    let count = new Array(10).fill(0);
    let set = new Set();
    let ans = 0;

    for (let digit of digits) {
        count[digit]++;
    }

    for (let i = 1; i <= 9; i++) {
        if (count[i] === 0) continue;

        // use i - means digit
        count[i]--;

        for (let j = 0; j <= 9; j++) {
            if (count[j] == 0) continue;

            // use j
            count[j]--;

            for (let k = 0; k <= 9; k++) {
                if (count[k] == 0) continue;

                // checks that the last digit should be even for even num
                if (k % 2 == 0) {

                    // use k
                    count[k]--;

                    // generate a num
                    let num = 100 * i + 10 * j + k;

                    if (!set.has(num)) {
                        set.add(num);
                        ans++;
                    }
                    // restore k
                    count[k]++;
                }
            }
            // restore j
            count[j]++
        }
        // restore i
        count[i]++;
    }
    return ans;
};